### DISCUSSION TITLE
shortcodes vs iframes for videos

### DISCUSSION THREAD

**Discussion by stuvjordan-uroc** [quarto-dev/quarto-cli#6501] [Q&A]
### Description

Hi!

I'm trying to embed a video following the docs here: https://quarto.org/docs/authoring/videos.html.

Here's the link to the video I'd like to embed: 

https://rochester.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=9577f0bf-065e-4611-8049-b05b01284126

(Panopto is the video hosting service my university provides for this sort of thing.)

When I do:

```
{{< video https://rochester.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=9577f0bf-065e-4611-8049-b05b01284126 >}}
```
rendering fails with the message "No video source specified for video shortcode"

Based on [this thread](https://github.com/quarto-dev/quarto-cli/discussions/3781), I'm guessing quarto doesn't like something about how the query parameter is formatted?  So I tried grabbing the URL returned when I ask Panopto to give me code to embed the video.  Using that url yields this:

```
{{< video https://rochester.hosted.panopto.com/Panopto/Pages/Embed.aspx?id=9577f0bf-065e-4611-8049-b05b01284126&autoplay=false&offerviewer=true&showtitle=true&showbrand=true&captions=false&interactivity=all >}}
```

This leads to successful compilation, but with a non-working result like this:

![image](https://github.com/quarto-dev/quarto-cli/assets/78857503/7dbd8bed-75b4-42d8-96a2-43ffdb7256f5)

Finally I tried just the the raw iframe code supplied by Panopto:

```
<iframe src="https://rochester.hosted.panopto.com/Panopto/Pages/Embed.aspx?id=9577f0bf-065e-4611-8049-b05b01284126&autoplay=false&offerviewer=true&showtitle=true&showbrand=true&captions=false&interactivity=all" height="405" width="720" style="border: 1px solid #464646;" allowfullscreen allow="autoplay"></iframe>
```

That works beautifully:

![image](https://github.com/quarto-dev/quarto-cli/assets/78857503/64f028d5-59ea-4e82-90dc-39ac1e41019a)

So what's going on here?  What are video shortcodes doing that is different from what raw iframe code does?  Am I missing out on something by not being able to make the shortcode work for this video?

Thanks!

This

**Comment by cscheid** [comment]
You can always check the HTML output that was generated by the shortcode and compare it to the raw `<iframe>` inclusion; maybe start there?
